planner/core/logical_plan_builder.go

```diff
@@ -1768,7 +1768,14 @@ func (b *PlanBuilder) checkOnlyFullGroupByWithGroupClause(p LogicalPlan, sel *as
 		}
 		switch errExprLoc.Loc {
 		case ErrExprInSelect:
-			return ErrFieldNotInGroupBy.GenWithStackByArgs(errExprLoc.Offset+1, errExprLoc.Loc, sel.Fields.Fields[errExprLoc.Offset].Text())
+			var errText string
+			selectField := sel.Fields.Fields[errExprLoc.Offset]
+			if colExpr, ok := selectField.Expr.(*ast.ColumnNameExpr); ok && len(schema.Columns) > 0 {
```

Comment:
Please consider other possible expressions like `Length(field)`

Reply:
Yes. and fixed by use a much more easier solution. and UT added